index.d.ts

```diff
@@ -115,16 +118,29 @@ export namespace Cmd {
     args?: any[]
   ): MapCmd<A>;
 
-  export function run<A extends Action, B extends Action>(
-    f: Function,
+  // Allow the use of special dispatch | getState symbols
+  type ArgOrSymbol<T> = {
+    [K in keyof T]: T[K] extends Dispatch | GetState
+      ? typeof dispatch | typeof getState | T[K]
+      : T[K];
+  }
+  export type PromiseResult<T> = T extends Promise<infer U> ? U : T;
+
+  export function run<
+    Err extends Error,
+    Func extends (...args: any[]) => Promise<any> | any,
+    SuccessAction extends Action,
+    FailAction extends Action,
+    >(
+    f: Func,
     options?: {
-      args?: any[];
-      failActionCreator?: ActionCreator<A>;
-      successActionCreator?: ActionCreator<B>;
+      args?: ArgOrSymbol<Parameters<Func>>;
+      failActionCreator?: (error: Err) => FailAction;
```

Comment:
A promise can fail with `error: any`. It is not always an `Error`. To be honest, I would leave this as `any`. In TypeScript Promises cannot specify the error type, so it might not be worth it to specify an error type here.

Reply:
I've updated this, it seems sensible